narwhals/_duckdb/dataframe.py

```diff
@@ -156,8 +156,13 @@ def select(
         )
 
     def drop(self: Self, columns: list[str], strict: bool) -> Self:  # noqa: FBT001
+        if strict:
+            return self._from_native_frame(
+                self._native_frame.select(duckdb.StarExpression(exclude=columns))
+            )
+
         columns_to_drop = parse_columns_to_drop(
-            compliant_frame=self, columns=columns, strict=strict
+            compliant_frame=self, columns=columns, strict=False
         )
```

Comment:
unrelated, but i had this in my local branch and figured i'd keep it, i prefer to use native apis where possible